### PR TITLE
[MSYS-661] [In-Progress]Fix chef_client_scheduled_task idempotence

### PR DESCRIPTION
**Pull request by dheerajd-msys**
### Description
Updated cookbook for making `chef_client_scheduled_task` resource idempotent.

### Issues Resolved
Fixes https://github.com/chef-cookbooks/chef-client/issues/490

### CODE REVIEW

**Comment by seththoenen** [comment]
Would the scheduled task resource update if someone wanted to change the functionality of the scheduled task? For example, would a new task be created if one wanted to change the interval or splay?

**Comment by dheerajd-msys** [comment]
Per current `chef-client` cookbook `chef_client_scheduled_task` resource will not update the interval or splay.
I have made it idempotent for only if `scheduled task` is created already.

**Comment by dheerajd-msys** [comment]
@btm @tas50 , Please review.

**Comment by tas50** [comment]
Won't this prevent the task from ever updating if the options or other properties change?

File: recipes/task.rb [hunk 1]
```diff
@@ -45,4 +45,5 @@ class ::Chef::Recipe
   log_directory node['chef_client']['log_dir']
   chef_binary_path node['chef_client']['bin']
   daemon_options node['chef_client']['daemon_options']
+  not_if { ::File.exist?("C:\\Windows\\System32\\Tasks\\#{node['chef_client']['task']['task_name']}") }
```
Comment (btm) [inline recipes/task.rb]: As @tas50 indicated, this would prevent the task from being updated.

Comment (dheerajd-msys) [reply recipes/task.rb]: Yes @btm, I've mentioned it too in my above comment. To make the resource `chef_client_scheduled_task` idempotent I was needed to add this check.

Comment (dheerajd-msys) [reply recipes/task.rb]: @tas50 @btm, This cookbook calculates the `start_time` for frequency `:minute` which seems inappropriate as per `windows_task` resource description. `start_time` needs to be mentioned only for `:once` 
```
frequency

    Ruby Type: Symbol | Default Value: :hourly

    Frequency with which to run the task. Values include: :minute, :hourly, :daily, :weekly, :monthly, :once, :on_logon, :onstart, :on_idle. The :once value requires the start_time property.
``` 
Please check here https://github.com/chef/chef-web-docs/blob/master/chef_master/source/resource_windows_task.rst#properties

I am trying to modify the recipe like gist shared here https://gist.github.com/dheerajd-msys/e771890bc8bf0f515582c77bba84c7ae.

Please suggest your views for modifications.

**Comment by NimishaS** [comment]
@tas50 , can you please confirm the reason of adding `start_time` in case of `minutes` frequency: https://github.com/chef-cookbooks/chef-client/blob/master/resources/scheduled_task.rb#L45? Does removing it has any bad impact?

We checked the commit history of the file, but the reason isn't very clear from the commit messages.

**Comment by dheerajd-msys** [comment]
We have identified a bug in windows_task resource for `frequency :monthly` that it is not idempotent. Also found that for `frequency :once` , `frequency_modifier` in `windows_task` option should not be provided. We will be fixing it soon on another PR so closing this now.

**Pull request closed without merging**